packages/token/src/tests/erc20/test_erc20.cairo

```diff
@@ -98,7 +98,8 @@ fn test_approve() {
     let mut spy = spy_events();
 
     start_cheat_caller_address(contract_address, OWNER());
-    assert!(state.approve(SPENDER(), VALUE));
+    let success = state.approve(SPENDER(), VALUE);
+    assert!(success, "Transfer failed");
```

Comment:
```suggestion
    assert!(success, "Approval failed");
```
Is this error message adding real value?

Reply:
It does once you have to debug the failing test

Reply:
You can go to the specific line from the test error message, and in this particular case, I think it is quite clear that that line error implies that the approval failed. How exactly does this help debug?

Reply:
Foundry output doesn't specify the number of the line that failed. However, it does write the variable names passed to the assertion and their values

For example, let's take the test cases for transfers. Both and are called in a single test cases and both helper functions contain the same assertion assert_eq!(initial_supply, current_supply);

How to find out which one has failed?

Reply:
Foundry output doesn't specify the number of the line that failed. However, it does write the variable names passed to the assertion and their values

For example, let's take the test cases for transfers. Both and are called in a single test cases and both helper functions contain the same assertion assert_eq!(initial_supply, current_supply);

How to find out which one has failed?

Reply:
I got your point, but I don't think "Approval failed" is going to help with that if we add the same message. Adding before and after (in the other examples) differentiates and is true it makes it easier to find, but if the message intention is only to find the line easier, we may directly put the line as a message, and I don't think the message should be used for that. We can/should push for snforge to print the line, but I'm still against using messages for this purpose.

Reply:
I see your point. Previously some error messages, like "Approval failed" here, did make it more obvious which exact assertion has failed. But after migration things got better since SnFoundry does a pretty good job printing the  assertion itself and the names of variables passed to it

I shall agree with you, most of the error messages can be safely omitted. In some cases like this, thought, we should keep them until SnFoundry starts printing the assertion's line number

Reply:
@ericnordelo @immrsd so I changed this back and removed the error messages because whether we _need_ these error messages or not for debugging is clearly debatable. Because it's debatable, I'm favoring consistency with how we've already used the `assert_state_x` assertions in the erc721 and erc1155 tests (with no error messages).

I propose we create a new issue and continue this discussion there. If we were to conclude that we should include error messages in these assertions, It'd be nice to also update the other token tests as well